lib/streamlit/runtime/app_session.py

```diff
@@ -632,14 +648,17 @@ def _create_file_change_message(self) -> ForwardMsg:
         msg.session_event.script_changed_on_disk = True
         return msg
 
-    def _create_new_session_message(self, page_script_hash: str) -> ForwardMsg:
+    def _create_new_session_message(
+        self, page_script_hash: str, fragment_id: Optional[str] = None
```

Comment:
nit: instead of `Optional[str]` it also can be typed via `str | None` via future annotations.